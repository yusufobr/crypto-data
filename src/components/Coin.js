import React from 'react'

function Coin() {
  return (
    <div>Coin</div>
  )
}

export default Coin