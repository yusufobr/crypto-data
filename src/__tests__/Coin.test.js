import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Coin from '../components/Coin';

class ResizeObserver {
  observe() {}

  unobserve() {}

  disconnect() {}
}

window.ResizeObserver = ResizeObserver;

const mockStore = configureStore([thunk]);

describe('Coin.js', () => {
  it('should render the Coin component', () => {
    const initialState = {
      coins: {
        coins: [
          {
            id: 'bitcoin',
            rank: '1',
            symbol: 'BTC',
            name: 'Bitcoin',
            supply: '19390006.0000000000000000',
            maxSupply: '21000000.0000000000000000',
            marketCapUsd: '523461027259.6814922831300980',
            volumeUsd24Hr: '3833686494.4517955591897860',
            priceUsd: '26996.4345168166266830',
            changePercent24Hr: '-0.3439222636312085',
            vwap24Hr: '26909.0357708240879410',
            explorer: 'https://blockchain.info/',
          },
        ],
      },
    };
    const store = mockStore(initialState);

    const tree = render(
      <Provider store={store}>
        <Router>
          <Coin />
        </Router>
      </Provider>,
    );

    expect(tree).toMatchSnapshot();
  });
});
