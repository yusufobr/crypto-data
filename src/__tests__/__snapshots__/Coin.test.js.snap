// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coin.js should render the Coin component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container p-2 mx-auto flex flex-col gap-4 text-white py-2 mb-8"
      >
        <button
          class="text-left"
          type="button"
        >
          <img
            alt="backarrow"
            class="p-2 bg-[#1dd79c] rounded-full"
            src="backArrow.svg"
            width="40"
          />
        </button>
        <div
          class="rounded-lg shadow-md bg-[#1E2423] pt-16 pb-4"
        >
          <div
            class=""
          >
            <div
              class="recharts-responsive-container"
              style="width: 95%; height: 300px; min-width: 0;"
            />
          </div>
          <div
            class="flex flex-col gap-2"
          >
            <span
              class="text-4xl uppercase font-extrabold text-center mt-4 text-[#161B19] "
            />
            <div>
              <ul
                class="flex flex-col text-lg myeven"
              >
                <li
                  class="flex justify-between items-center"
                >
                  <span>
                    Symbol
                  </span>
                  <span />
                </li>
                <li
                  class="flex justify-between items-center"
                >
                  <span>
                    Price USD
                  </span>
                  <span
                    class="font-semibold"
                  >
                    NaN $
                  </span>
                </li>
                <li
                  class="flex justify-between items-center"
                >
                  <span>
                    Supply
                  </span>
                  <span>
                    NaN $
                  </span>
                </li>
                <li
                  class="flex justify-between items-center"
                >
                  <span>
                    Max Supply USD
                  </span>
                  <span>
                    Not Available
                  </span>
                </li>
                <li
                  class="flex justify-between items-center"
                >
                  <span>
                    Market Capital USD
                  </span>
                  <span>
                    Not Available $
                  </span>
                </li>
                <li
                  class="flex justify-between items-center"
                >
                  <span>
                    Volume 24h USD
                  </span>
                  <span>
                    Not Available $
                  </span>
                </li>
                <li
                  class="flex justify-between items-center"
                >
                  <span>
                    Rank
                  </span>
                  <span />
                </li>
                <li
                  class="flex justify-between items-center"
                >
                  <span>
                    Last 24h Change
                  </span>
                  <span
                    class="text-green-500"
                  >
                    Not Available %
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </div>
        <a
          class="text-right pt-4"
          rel="noopener noreferrer"
          target="_blank"
          type="button"
        >
          <span
            class="py-2 text-black px-8 bg-[#1dd79c] rounded"
          >
            Website
          </span>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container p-2 mx-auto flex flex-col gap-4 text-white py-2 mb-8"
    >
      <button
        class="text-left"
        type="button"
      >
        <img
          alt="backarrow"
          class="p-2 bg-[#1dd79c] rounded-full"
          src="backArrow.svg"
          width="40"
        />
      </button>
      <div
        class="rounded-lg shadow-md bg-[#1E2423] pt-16 pb-4"
      >
        <div
          class=""
        >
          <div
            class="recharts-responsive-container"
            style="width: 95%; height: 300px; min-width: 0;"
          />
        </div>
        <div
          class="flex flex-col gap-2"
        >
          <span
            class="text-4xl uppercase font-extrabold text-center mt-4 text-[#161B19] "
          />
          <div>
            <ul
              class="flex flex-col text-lg myeven"
            >
              <li
                class="flex justify-between items-center"
              >
                <span>
                  Symbol
                </span>
                <span />
              </li>
              <li
                class="flex justify-between items-center"
              >
                <span>
                  Price USD
                </span>
                <span
                  class="font-semibold"
                >
                  NaN $
                </span>
              </li>
              <li
                class="flex justify-between items-center"
              >
                <span>
                  Supply
                </span>
                <span>
                  NaN $
                </span>
              </li>
              <li
                class="flex justify-between items-center"
              >
                <span>
                  Max Supply USD
                </span>
                <span>
                  Not Available
                </span>
              </li>
              <li
                class="flex justify-between items-center"
              >
                <span>
                  Market Capital USD
                </span>
                <span>
                  Not Available $
                </span>
              </li>
              <li
                class="flex justify-between items-center"
              >
                <span>
                  Volume 24h USD
                </span>
                <span>
                  Not Available $
                </span>
              </li>
              <li
                class="flex justify-between items-center"
              >
                <span>
                  Rank
                </span>
                <span />
              </li>
              <li
                class="flex justify-between items-center"
              >
                <span>
                  Last 24h Change
                </span>
                <span
                  class="text-green-500"
                >
                  Not Available %
                </span>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <a
        class="text-right pt-4"
        rel="noopener noreferrer"
        target="_blank"
        type="button"
      >
        <span
          class="py-2 text-black px-8 bg-[#1dd79c] rounded"
        >
          Website
        </span>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
